template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java

```diff
@@ -22,13 +25,31 @@ public class HdfsRoleConfigProvider extends AbstractRoleConfigProvider {
 
     private static final Integer NUM_FAILED_VOLUMES_TOLERATED = 0;
 
+    private static final String DFS_REPLICATION = "dfs_replication";
+
+    private static final Integer DFS_REPLICATION_VALUE = 2;
+
+    private final EntitlementService entitlementService;
+
+    public HdfsRoleConfigProvider(EntitlementService entitlementService) {
+        this.entitlementService = entitlementService;
+    }
+
     @Override
     protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
         switch (roleType) {
             case HdfsRoles.DATANODE:
-                return List.of(
+                List<ApiClusterTemplateConfig> configs = new ArrayList<>();
+                configs.add(
                         config(FAILED_VOLUMES_TOLERATED, NUM_FAILED_VOLUMES_TOLERATED.toString())
                 );
+                if (entitlementService.isSDXOptimizedConfigurationEnabled(ThreadBasedUserCrnProvider.getAccountId())
+                && isNamenodeHA(source)) {
```

Comment:
How are we differentiating here that this is a config is only meant for a DL?
Wouldn't this mean that if we have this entitlement we'll have the max replication count set to 2?

Reply:
I agree. This change should be limited to datalake. 
The current change will limit the replication count to 2 for any cluster that had HDFS.

Reply:
Add the below condition to limit it to the datalake cluster.

source.getStackType().equals(StackType.DATALAKE)

Reply:
resolved